parser.go

```diff
@@ -1281,21 +1281,55 @@ func defineTypeOfExample(schemaType, arrayType, exampleValue string) (interface{
 	return nil, fmt.Errorf("%s is unsupported type in example value %s", schemaType, exampleValue)
 }
 
-// GetAllGoFileInfo gets all Go source files information for given searchDir.
-func (parser *Parser) getAllGoFileInfo(packageDir, searchDir string) error {
-	return filepath.Walk(searchDir, func(path string, f os.FileInfo, _ error) error {
-		if err := parser.Skip(path, f); err != nil {
+// GetAllGoFileInfoAndParseTypes gets all Go source files information for given searchDir and parses the types from them.
+func (parser *Parser) GetAllGoFileInfoAndParseTypes(searchDir string) error {
```

Comment:
I need this in order to be able to go through packages one by one and incrementally parse the types from them.

I saw that ParseAPIMultiSearchDir() first collects all types from the astFiles and then parses them all at once via parser.packages.ParseTypes(), but in my case that would mean having to do a double pass through all routes, which I'd rather not have to do and instead have a function that parses the types of a single package.